account/views.py

```diff
@@ -1,21 +1,37 @@
-from rest_framework import viewsets
+from rest_framework import viewsets, mixins
 from rest_framework.permissions import IsAuthenticated
-
+from rest_framework.response import Response
+import json
 
 from account.models import (
+    Note,
     Admin,
     Student,
     Parent,
     Instructor
 )
 from account.serializers import (
+    NoteSerializer,
     AdminSerializer,
     StudentSerializer,
     ParentSerializer,
     InstructorSerializer
-)
-
-
+) 
+
+class NoteViewSet(mixins.ListModelMixin, mixins.RetrieveModelMixin, mixins.CreateModelMixin, mixins.UpdateModelMixin, mixins.DestroyModelMixin, viewsets.GenericViewSet):
+    queryset = Note.objects.all()
+    serializer_class = NoteSerializer
+
+    def list(self, request):
+        """
+        override list to return notes for specific user
+        """
+        data = json.loads(request.body)
```

Comment:
GET requests should not have a body. we should not be parsing the request body here.

The standard way to do this is to get the value from the query params in the URL:
`self.request.query_params.get("user_id", None)`
and throw an error if this doesn't exist.

See this link for an example: https://harrymoreno.com/2019/06/12/Overriding-Django-Rest-Framework-viewsets.html